app/models/Language.py

```diff
@@ -0,0 +1,78 @@
+from misc import lang_path
+from json import loads as json_loads
+from hashlib import md5
+
+# JSON of languages
+with open(lang_path) as lang_data:
+    languages = json_loads(lang_data.read())
+
+class Language(object):
+    """
+    Supplies all known information about a language. Make an instance using the
+    language id. If a language does not exist then this will *not* error however
+    some functions will return `None` appropriately.
+    """
+    def __init__(self, lang_id):
+        self._id = Language.normalize(lang_id)
+
+        # Should not fail otherwise you have malformed source code
+        self._data = languages['languages'][self._id]
+
+    def get_color(self):
+        return '#' + "".join(md5(self._id.encode('ascii')).hexdigest()[:6])
+
+    def get_short_id(self):
+        second_char = self._id[1] if len(self._id) > 1 else ""
+        return self._data.get('sn', self._id[0].upper() + second_char)
+
+    def get_display_name(self):
+        # Return display name. If that doesn't exist, capitalize each letter
+        # following a whitespace char
+
+        if self._data is None:
+            return None
+
+        return self._data.get('display', this._id.title())
+
+    def to_json(self):
+        """
+        May not return a JSON object
+        """
+        return self._id
+
+    def get_id(self):
```

Comment:
This isn't Java. Python attributes don't need getters.

Reply:
I added get_id as I a not sure `_id` will always be the resolved id of the language. It might be a capitalized, or alias id.